selfhost/repl.jakt

```diff
@@ -31,12 +31,54 @@ struct Editor {
             throw Error::from_errno(42)
         }
 
-        let editor = Editor(
+        return Editor(
             standard_input_file: std_in
             line_pointer: allocate<c_char>(count: 4096)
             prompt
         )
-        return editor
+    }
+
+    fn check_parens(anon input: String) -> bool {
+        mut unmatched_parens   = 0
+        mut unmatched_brackets = 0
+        mut unmatched_curlies  = 0
+
+        mut pos: usize = 0
+
+        while pos < input.length() {
+            match input.byte_at(pos) {
```

Comment:
This will break with parens/brackets/curlies inside strings, that's why I suggested using the lexer